lib/index.coffee

```diff
@@ -0,0 +1,58 @@
+_ = require "underscore"
+_.mixin require 'underscore.deep'
+async = require "async"
+genericPool = require "generic-pool"
+
+create_pool = (thrift, options) ->
+  genericPool.Pool
+    name: "thrift"
+    create: (cb) ->
+      connection = thrift.createConnection options.host, options.port, {timeout: options.timeout}
+      connection.__ended = false
+      connection.on "connect", ->
+        connection.connection.setKeepAlive(true)
+        cb null, connection
+      connection.on "close", ->
+        connection.__ended = true
+      connection.on "error", (err) ->
+        connection.__ended = true
+        cb err
```

Comment:
@kvigen @jonahkagan added additional tests and changed slight behavior of index to take care of this issue.
f03c6b20fe76015af0cf1037acc7d40518b166ac